File: Question/Question 21-30/26. Remove Duplicates from Sorted Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
let removeDuplicates = function(nums) {
    let index = 1;
    for(let i = 0; i < nums.length-1; i++){
        if(nums[i] != nums[i+1]){
            nums[index++] = nums[i+1];
        }
    }
};

var removeDuplicates1 = function(nums) {
    let index = 1
    for(let i = 0 ;i<nums.length;i++){
        if(nums[i]!==nums[i+1]){
            nums[index++] = nums[i+1]
        }
    }
    return index-1
};

File: DailyQuestion/java/April/26. Remove Duplicates from Sorted Array.java
package DailyQuestion.java.April;
class Solution {
  public int removeDuplicates(int[] nums) {
      if(nums == null || nums.length == 0){
          return 0;
      }
      int index=1;
      for(int i=0; i<nums.length - 1; i++){
          if(nums[i] != nums[i+1]){
              if(index != i+1){//避免原地复制
                  nums[index] = nums[i+1];
              }
              index++;
          }
      }
      return index;
  } 
}